be/src/vec/functions/function_string.cpp

```diff
@@ -243,6 +243,156 @@ struct StringFunctionImpl {
+        
+        auto rows_count = offsets.size();
+        // allocate memory and memory align
+        res_offsets.resize(rows_count);
+        
+        // offsets[-1]=0,equals assign 0.
+        auto pre_offset = offsets[-1];
+        for (size_t i = 0; i < rows_count; ++i) {
+            const char* src_str = reinterpret_cast<const char*>(&data[offsets[i - 1]]);
+            //use char*
+            char* dst_str = reinterpret_cast<char*>(&res_data[res_offsets[i - 1]]);
+            
+            fast_hex_encode(src_str, dst_str, "0123456789ABCDEF");
+            
+            // sum use to get count of every char size of src_raw_str
+            auto sum = 0;
+            for (size_t j = 0, char_size = 0; j < strlen(dst_str); j += char_size) {
+                char_size = get_utf8_byte_length((unsigned) (dst_str)[j]);
+                sum += char_size;
+            }
+            // change res_offsets
+            res_offsets[i - 1] = pre_offset;
+            // 1 means '\0'
+            res_offsets[i] = res_offsets[i - 1] + sum + 1;
+            pre_offset = res_offsets[i];
+        }
+        return Status::OK();
+    }
+};
+
+struct NameUnHex{
+    static constexpr auto name = "unhex";
+};
+struct UnHexImpl{
+    static bool check_and_decode_one(char& c, const char src_c, bool flag) {
+        int k = flag ? 16 : 1;
+        int value = src_c - '0';
+        // 9 = ('9'-'0')
+        if (value >= 0 && value <= 9) {
+            c += value * k;
+            return true;
+        }
+        
+        value = src_c - 'A';
+        // 5 = ('F'-'A')
+        if (value >= 0 && value <= 5) {
+            c += (value + 10) * k;
+            return true;
+        }
+        
+        value = src_c - 'a';
+        // 5 = ('f'-'a')
+        if (value >= 0 && value <= 5) {
+            c += (value + 10) * k;
+            return true;
+        }
+        // not in ( ['0','9'], ['a','f'], ['A','F'] )
+        return false;
+    }
+    
+    static void fast_hex_decode(const char* src_str, char* dst_str) {
+        
+        // if str length is odd or 0 ,return src_str like mysql dose.
+        auto src_len = strlen(src_str);
+        if ((src_len & 1) != 0 or src_len == 0) {
+            std::copy(src_str, src_str + src_len, dst_str);
+            *(dst_str + src_len) = '\0';
+            return;
+        }
+        //check and decode one character at the same time
+        // character in ( ['0','9'], ['a','f'], ['A','F'] ), return 'NULL' like mysql dose.
+        for (auto i = 0, dst_index = 0; i < src_len; i += 2, dst_index++) {
+            char c = 0;
+            // combine src_str two character into dst_str one character , bool param means weather multiple 16.
+            bool left_4bits_flag = check_and_decode_one(c, *(src_str + i), true);
+            bool right_4bits_flag = check_and_decode_one(c, *(src_str + i + 1), false);
+            
+            if (!left_4bits_flag || !right_4bits_flag) {
+                *dst_str = 'N';
```

Comment:
I think returning null means marking a NullMap in a nullable column, not a string with a value of `NULL`.

Reply:
ok, i will change it.